### PR TITLE
Check cpu info valid during runtime

### LINKED ISSUE #6075: v6.3.0 tiflash coredump when running in a CPU without AVX support
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Running v6.3.0 on a server with a CPU without AVX support

```
$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    2
Socket(s):             2
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            15
Model:                 6
Model name:            Common KVM processor
Stepping:              1
CPU MHz:               2199.998
BogoMIPS:              4399.99
Hypervisor vendor:     KVM
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              4096K
L3 cache:              16384K
NUMA node0 CPU(s):     0-3
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx lm constant_tsc nopl xtopology eagerfpu pni cx16 x2apic hypervisor lahf_lm
```

### 2. What did you expect to see? (Required)

tiflash starts successfully

### 3. What did you see instead (Required)

TiFlash coredump because running into an AVX instruction `vmovups`. However, the CPU does not support AVX instructions.
Introduced by https://github.com/pingcap/tiflash/pull/5658. Ref https://github.com/pingcap/tiflash/issues/5294

```
$ ./tiflash version
Illegal instruction (core dumped)

$ gdb --args ./tiflash version
...
(gdb) bt
#0  0x0000000001be0193 in _GLOBAL__sub_I_CollationStringOptimized.cpp ()
#1  0x0000000001bd993d in __libc_csu_init ()
#2  0x00007ffff35424e5 in __libc_start_main () from /lib64/libc.so.6
#3  0x0000000001a95f69 in _start ()
(gdb) disassemble
Dump of assembler code for function _GLOBAL__sub_I_CollationStringOptimized.cpp:
...
   0x0000000001be0187 <+87>:    callq  0x7c13a10 <_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEC1ERKS5_@plt>
   0x0000000001be018c <+92>:    movb   $0x20,0x60583dd(%rip)        # 0x7c38570 <_ZN2DB6Errors11PageStorageL16FileSizeNotMatchE.434+24>
=> 0x0000000001be0193 <+99>:    vmovups -0x1968f28(%rip),%xmm0        # 0x277273
   0x0000000001be019b <+107>:   vmovups %xmm0,0x60583ce(%rip)        # 0x7c38571 <_ZN2DB6Errors11PageStorageL16FileSizeNotMatchE.434+25>
   0x0000000001be01a3 <+115>:   movb   $0x0,0x60583d7(%rip)        # 0x7c38581 <_ZN2DB6Errors11PageStorageL16FileSizeNotMatchE.434+41>
   0x0000000001be01aa <+122>:   mov    $0x30,%edi
```

### 4. What is your TiFlash version? (Required)
v6.3.0

v6.2.0 can run successfully with this CPU

### PR DESCRIPTION
**Pull request by solotzg**
### What problem does this PR solve?

Issue Number: close #6075

- mongodb 5.0(since Jul 13, 2021) requires use of the AVX instruction set.
- starrocks declares that CPU must support AVX2 instruction sets.
- apache doris provide avx2(default) and optional no-avx2 version since 1.1.0(released on 2022-07-14).

For tiflash, it's time to make avx2 as default required CPU feature.

### What is changed and how it works?

- TiFlash requires CPU support AVX2 instruction sets if `TIFLASH_ENABLE_AVX_SUPPORT` is used for building.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Common/TiFlashBuildInfo.cpp [hunk 1]
```diff
@@ -133,6 +135,22 @@ std::string getProfile()
...
+    // CPU which supports avx2 must have supported sse4, sse...
+    valid &= common::cpu_feature_flags.sse4_1 && common::cpu_feature_flags.sse4_2;
+    valid &= common::cpu_feature_flags.popcnt;
+    if (!valid)
+    {
+        return "Tilash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe";
+    }
+#endif
+    return {};
+}
```
[the remark this answers is not in the document]
Comment (solotzg) [reply dbms/src/Common/TiFlashBuildInfo.cpp]: We may still need to check other flags.

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Common/TiFlashBuildInfo.cpp [hunk 2]
```diff
@@ -133,6 +135,29 @@ std::string getProfile()
...
+        // sse4.1 sse4.2 popcnt are required in `cmake/test_cpu.cmake`
+        & common::cpu_feature_flags.popcnt
+        // movbe is necessary for memcmp
+        & common::cpu_feature_flags.movbe
+        //
+        ;
+
+    if (!valid)
+    {
+        return "Tilash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe. Please check `/proc/cpuinfo`.";
```
Comment (solotzg) [inline dbms/src/Common/TiFlashBuildInfo.cpp:155]:
```suggestion
        return "TiFlash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe. Please check CPU info(`/proc/cpuinfo` under Linux).";
```

File: dbms/src/Server/main.cpp [hunk 3]
```diff
@@ -150,6 +150,12 @@ bool isClickhouseApp(const std::string & app_suffix, std::vector<char *> & argv)
 
 int main(int argc_, char ** argv_)
 {
+    if (auto r = TiFlashBuildInfo::CheckRuntimeValid(); r)
+    {
+        std::cerr << *r << std::endl;
+        exit(-1);
+    }
```
Comment (JaySon-Huang) [inline dbms/src/Server/main.cpp:157]:
```
$ ./tiflash version
Illegal instruction (core dumped)

$ gdb --args ./tiflash version
...
(gdb) bt
#0  0x0000000001be0193 in _GLOBAL__sub_I_CollationStringOptimized.cpp ()
#1  0x0000000001bd993d in __libc_csu_init ()
#2  0x00007ffff35424e5 in __libc_start_main () from /lib64/libc.so.6
#3  0x0000000001a95f69 in _start ()
```
TiFlash will coredump before entering `main`, I think this check doesn't help...

Comment (solotzg) [reply dbms/src/Server/main.cpp:157]: After this PR, if env does not supoort required CPU instruction sets, TiFlash will cout err message and exit -1.
```
$ ./tiflash --version                                  
TiFlash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe. Please check CPU info(`/proc/cpuinfo` under Linux).

$ ./tiflash server --config xxx
TiFlash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe. Please check CPU info(`/proc/cpuinfo` under Linux).
```

Comment (solotzg) [reply dbms/src/Server/main.cpp:157]: I see. We may need to check CPU info while deploying.

**Comment by solotzg** [comment]
Check required CPU flags for TiFlash in https://github.com/pingcap/tiup/pull/2054. Close thit PR.

**Pull request closed without merging**